torch/fx/experimental/rewriter.py

```diff
@@ -65,28 +65,62 @@ def visit_Assert(self, node):
         return ast.copy_location(expr_wrapper, node)
 
 
-class RewritingTracer(Tracer):
+class AssertRewritingTracer(Tracer):
     def trace(self, root: Union[torch.nn.Module, Callable], concrete_args: Optional[Dict[str, Any]] = None) -> Graph:
-        return super().trace(_rewrite(root), concrete_args)
+        foo = _rewrite(root)
+        return super().trace(foo, concrete_args)
 
 
 def _rewrite(fn: Union[torch.nn.Module, Callable]) -> Union[torch.nn.Module, Callable]:
+
     if isinstance(fn, torch.nn.Module):
-        # Rewrite this module's `forward` as well as the `forward`s of
-        # all of this module's recursive descendents. Return the new,
-        # rewritten module hierarchy.
+
+
         def rewrite_module(m : torch.nn.Module):
+
             class RewrittenModule(torch.nn.Module):
                 def __init__(self, orig):
                     super().__init__()
+
+                    def is_helper_method(k: str) -> bool:
+
+                        def is_dunder(name: str) -> bool:
+                            return (len(name) > 4
+                                    and name.startswith('__') 
+                                    and name.endswith('__'))
+
+                        o = getattr(orig, k)
+
+                        if not callable(o):
+                            return False
+
+                        if is_dunder(k):
+                            return False
+
+                        if isinstance(o, torch.nn.Module):
+                            return False
+
+                        if k in dir(self):
+                            return False
+
+                        return True
+
+                    # Copy over any potential helper methods
+                    for k in dir(orig):
+                        if is_helper_method(k):
+                            self.__dict__[k] = getattr(orig, k)
+
```

Comment:
I wonder if this could be done easier and more resiliently by just making `RewrittenModule` subclass `type(m)`. Then everything on the class will be carried over and you can install a new `forward` on `RewrittenModule` without disturbing the original `Module`

Reply:
Maybe something like

```
class RewrittenModule(type(m)):
  pass
rewritten = copy.copy(m)
rewritten.__class__ = RewrittenModule

RewrittenModule.forward = ...
```